packages/jest/src/migrations/update-8-10-0/update-8-10-0.ts

```diff
@@ -0,0 +1,129 @@
+import {
+  chain,
+  Rule,
+  SchematicContext,
+  Tree
+} from '@angular-devkit/schematics';
+import {
+  formatFiles,
+  insert,
+  readWorkspace,
+  updatePackagesInPackageJson
+} from '@nrwl/workspace';
+import * as ts from 'typescript';
+import * as path from 'path';
+import {
+  Change,
+  getSourceNodes,
+  InsertChange,
+  readJsonInTree,
+  ReplaceChange
+} from '@nrwl/workspace/src/utils/ast-utils';
+import { stripIndents } from '@angular-devkit/core/src/utils/literals';
+
+export default function update(): Rule {
+  return chain([
+    displayInformation,
+    updatePackagesInPackageJson(
+      path.join(__dirname, '../../../', 'migrations.json'),
+      '8.10.0'
+    ),
+    updateJestConfigs,
+    formatFiles()
+  ]);
+}
+
+function displayInformation(host: Tree, context: SchematicContext) {
+  const config = readJsonInTree(host, 'package.json');
+  if (config.devDependencies && config.devDependencies['jest-preset-angular']) {
+    context.logger.info(stripIndents`
+    \`jest-preset-angular\` 8.0.0 has restructured folders, introducing breaking changes to 
+    jest.config.js files.
+    
+    We are updating snapshotSerializers in each Angular project to include appropriate paths.
+    
+    See: https://github.com/thymikee/jest-preset-angular/releases/tag/v8.0.0
+  `);
+  }
+}
+
+function updateJestConfigs(host: Tree) {
+  const config = readJsonInTree(host, 'package.json');
+
+  if (config.devDependencies && config.devDependencies['jest-preset-angular']) {
+    const workspaceConfig = readWorkspace(host);
+    const jestConfigsToUpdate = [];
+
+    Object.keys(workspaceConfig.projects).forEach(name => {
+      const project = workspaceConfig.projects[name];
```

Comment:
We should go through all of the architect targets.. for targets which use `@nrwl/jest:jest` as the builder... and use the `jestConfig` option verbatim if it exists so that we can catch all `jest.config.js` which are referenced in `workspace.json`.

Reply:
I don't get your point. It something not correct here?

Reply:
Sorry, let me clarify, this only migrates targets which have a `test` target that uses `@nrwl/jest:jest`.

A user could have the following:

```
projects: {
  app1: {
    architect: {
      e2e: {
        builder: '@nrwl/jest:jest',
        options: { ... }
      }
    }
  }
```

Such projects should be migrated as well.